src/main/scala/is/hail/expr/ir/functions/StringFunctions.scala

```diff
@@ -104,5 +104,27 @@ object StringFunctions extends RegistryFunctions {
 
       EmitTriplet(setup, missing, value)
     }
+
+    registerCode("hamming", TString(), TString(), TInt32()) { case (mb: EmitMethodBuilder, v1: Code[Long], v2: Code[Long]) =>
+      val len = mb.newLocal[Int]
+      val i = mb.newLocal[Int]
+      val n = mb.newLocal[Int]
+      val region: Code[Region] = mb.getArg[Region](1)
+
+      Code(
+        len := TBinary.loadLength(region, v1),
+        len.cne(TBinary.loadLength(region, v2)).mux(
+          Code._fatal("function 'hamming' requires strings to have equal length."),
```

Comment:
I'm coming around to the idea that we should use as few fatals as possible. Should we just return NA if this assertion fails? I'm not sure, but maybe that's nicer.

Reply:
It doesn't super matter for the alt allele use case. do we expect it to be used other places where we'd want a fatal to be thrown? I'm happy to change the behavior here if you want.